nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/main/java/org/apache/nifi/processors/influxdb/ExecuteInfluxDBQuery.java

```diff
@@ -0,0 +1,258 @@
+
+            if ( incomingFlowFile == null && context.hasNonLoopConnection() ) {
+                return;
+            }
+
+            if ( incomingFlowFile.getSize() == 0) {
+                getLogger().error("Empty query");
+                incomingFlowFile = session.putAttribute(incomingFlowFile, INFLUX_DB_ERROR_MESSAGE, "Empty query size is " + incomingFlowFile.getSize());
+                session.transfer(incomingFlowFile, REL_FAILURE);
+                return;
+            }
+
+            charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions(incomingFlowFile).getValue());
+            try {
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                session.exportTo(incomingFlowFile, baos);
+                baos.close();
+                query = new String(baos.toByteArray(), charset);
+            } catch(IOException ioe) {
+                getLogger().error("Exception while reading from FlowFile " + ioe.getLocalizedMessage(), ioe);
+                throw new ProcessException(ioe);
+            }
+
+            database = context.getProperty(DB_NAME).evaluateAttributeExpressions(incomingFlowFile).getValue();
+            queryResultTimeunit = TimeUnit.valueOf(context.getProperty(INFLUX_DB_QUERY_RESULT_TIMEUNIT).evaluateAttributeExpressions(incomingFlowFile).getValue());
+
+            outgoingFlowFile = incomingFlowFile;
+        } else {
+            charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions().getValue());
+            query = context.getProperty(INFLUX_DB_QUERY).evaluateAttributeExpressions().getValue();
+            database = context.getProperty(DB_NAME).evaluateAttributeExpressions().getValue();
+            queryResultTimeunit = TimeUnit.valueOf(context.getProperty(INFLUX_DB_QUERY_RESULT_TIMEUNIT).evaluateAttributeExpressions().getValue());
+
+            outgoingFlowFile = session.create();
+        }
+
+        try {
+            long startTimeMillis = System.currentTimeMillis();
+            QueryResult result = executeQuery(context, database, query, queryResultTimeunit);
+
+            if ( result == null ) {
+                String message = "Query Result was " + result;
+                getLogger().error(message);
+                throw new NullPointerException(message);
+            }
+
+            String json = gson.toJson(result);
+
+            if ( getLogger().isDebugEnabled() ) {
+                getLogger().debug("Query result {} ", new Object[] {result});
+            }
+
+            ByteArrayInputStream bais = new ByteArrayInputStream(json.getBytes(charset));
+            session.importFrom(bais, outgoingFlowFile);
+            bais.close();
+
+            final long endTimeMillis = System.currentTimeMillis();
+
+            if ( ! result.hasError() ) {
+                outgoingFlowFile = session.putAttribute(outgoingFlowFile, INFLUX_DB_EXECUTED_QUERY, String.valueOf(query));
```

Comment:
You might want to put this on the failure scenario too because there could be syntax errors getting reported. This is especially going to be helpful if the user is dynamically generating the query.

Reply:
If there is any error the `hasError` returns true and the flow file is transferred to success.  Let me know if I've missed anything.